### PR TITLE
[FLINK-11227] [table] The DescriptorProperties contains some bounds checking errors

### PR DESCRIPTION
**Pull request by xccui**
## What is the purpose of the change

This PR aims to fix the bounds checking errors for `validateFixedIndexedProperties()` and `validateArray()` in `DescriptorProperties`.

## Brief change log

Change the upper bounds of the loops from `< maxIndex` to `<= maxIndex`.

## Verifying this change

1. To verify `validateArray()`, made the last element an invalid one.
2. To verify `validateFixedIndexedProperties()`, added a new test case `validateFixedIndexedProperties()`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/descriptors/DescriptorPropertiesTest.scala [hunk 1]
```diff
@@ -165,4 +182,26 @@ class DescriptorPropertiesTest {
       minLength,
       maxLength)
   }
+
+  private def testFixedFieldsValidation(properties: DescriptorProperties): Unit = {
```
Comment (hequn8128) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/descriptors/DescriptorPropertiesTest.scala]: How about rename `testFixedFieldsValidation` to `testFixedIndexedPropertiesValidation`?

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/descriptors/DescriptorPropertiesTest.scala [hunk 2]
```diff
@@ -118,6 +122,19 @@ class DescriptorPropertiesTest {
     testArrayValidation(properties, 1, Integer.MAX_VALUE)
   }
 
+  @Test(expected = classOf[ValidationException])
+  def testInvalidFixedIndexedProperty(): Unit = {
```
Comment (hequn8128) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/descriptors/DescriptorPropertiesTest.scala]: Properties

**Comment by xccui** [comment]
Thanks for the review @hequn8128. The methods are renamed according to your comments.

Best, Xingcan

**Comment by hequn8128** [comment]
@xccui  Thanks a lot for the update. +1 to merge.

Best, Hequn

**Pull request closed without merging**